runtime/scheduler.cpp

```diff
@@ -71,19 +75,25 @@ static unsigned int get_rand(unsigned int state) {
     return state >> 16;
 }
 
-static void worker_change_state(__cilkrts_worker *w,
-                                enum __cilkrts_worker_state s) {
+void local_state::change_state(enum __cilkrts_worker_state s) {
     /* TODO: Update statistics based on state change. */
-    CILK_ASSERT(w->l->state != s);
-    w->l->state = s;
-#if 0
-    /* This is a valuable assertion but there is no way to make it
-       work.  The reducer map is not moved out of the worker until
-       long after scheduler state has been entered.  */
-    if (s != WORKER_RUN) {
-	CILK_ASSERT(w->reducer_map == NULL);
-    }
-#endif
+    CILK_ASSERT(state != s);
+    state = s;
+}
+
+// JFC: The following comment refers to Cilk-M, not OpenCilk.
```

Comment:
Good point.  Can we simply delete the comment, or is there a similar subtlety in OpenCilk's synchronization?